scalameter-core/src/main/scala/org/scalameter/Context.scala

```diff
@@ -20,6 +20,9 @@ case class Context(properties: immutable.Map[Key[_], Any]) {
     }
   }
   def goe[T](key: Key[T], v: => T) = properties.getOrElse(key, v).asInstanceOf[T]
+  def apply[T](key: KeyWithDefaultKey[T]) = {
```

Comment:
Well, I guess that this means that I should either uses intersection type or change `apply` signature to:
```
def apply[T](key: Key[T])
```
and throw error on runtime if `key` is not `KeyWithDefault` or `KeyWithDefaultKey`
What solution do you prefer?